### PR TITLE
Brokers can update 'started/done/pending' on inquiries 

### PR DESCRIPTION
**Pull request by arunbhalli**
[**PT-Story:**  ](https://www.pivotaltracker.com/story/show/178423919)

### User Story 
``` 
As a broker
In order to maintain and keep Inquiries up to date
I'd like to see and update Inquiry's status
``` 
## Changes proposed in this pull request: 
- Add a section with radio boxes to the collapse
- PUT request to change status of Inquiry
- Add filter buttons for status
- Adds environmental dev / prod stages

## What I have learned working on this feature: 
* useRef hook
* update a fixture in Cypress(!)
* Homemade filter system
* Set environment stage in React and use this in conditionals

## Screenshots (Client) 
https://recordit.co/HTFADxvatM

### CODE REVIEW

File: cypress/integration/brokerIsAbleToUpdateStatusOfInquiry.feature.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        'https://flex-coast-development.herokuapp.com/api/inquiries/**',
+        {
+          message: 'The inquiry status has been updated',
+        }
+      )
+    })
+    it('is expected to update the status to started', () => {
+      cy.get('[data-cy=inquiry-collapsible-cell]').within(() => {
+        cy.fixture('listOfInquiries.json').then((fixture) => {
+          fixture.inquiries[0].inquiry_status = 'started'
```
Comment (DavDan1) [inline cypress/integration/brokerIsAbleToUpdateStatusOfInquiry.feature.js:26]: is this like just for picking the [0] kind of first one in fixture?

File: src/components/StatusFilter.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const [open, setOpen] = useState(false)
+  const buttonEl = useRef(null)
+
+  const handleStatusChange = (event) => {
+    setStatus({ ...status, [event.target.name]: event.target.checked })
+  }
+
+  return (
+    <>
+      <IconButton ref={buttonEl} onClick={() => setOpen(true)}>
```
Comment (DavDan1) [inline src/components/StatusFilter.jsx:19]: what is ref={buttonEl}

File: src/components/StatusFilter.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import FilterListIcon from '@material-ui/icons/FilterList'
+import Popover from '@material-ui/core/Popover'
+import Checkbox from '@material-ui/core/Checkbox'
+import FormGroup from '@material-ui/core/FormGroup'
+import FormControlLabel from '@material-ui/core/FormControlLabel'
+import IconButton from '@material-ui/core/IconButton'
+
+const StatusFilter = ({ status, setStatus }) => {
+  const [open, setOpen] = useState(false)
+  const buttonEl = useRef(null)
```
Comment (christerforsgren91) [inline src/components/StatusFilter.jsx:11]: haven't used this hook before, why we using this? :)

**Review by DavDan1 — APPROVED** [review]

File: src/components/InquiryRows.jsx [hunk 4]
```diff
@@ -66,6 +81,38 @@ const InquiryRows = ({ item }) => {
...
+                      <FormControlLabel
+                        data-cy='status-btn-3'
+                        value='done'
+                        control={<Radio />}
+                        label='Done'
+                      />
+                    </RadioGroup>
+                  </FormControl>
+                </div>
+                {error && <ErrorMessage text={message} />}
```
Comment (offcast-dev) [inline src/components/InquiryRows.jsx:115]: Material UI has this component that can be cool for errors. https://material-ui.com/components/snackbars/
So the table can say something like 'No inquiries yet' but the actual 500 error txt can be in snackbar on top of the screen

File: cypress/integration/brokerIsAbleToUpdateStatusOfInquiry.feature.js (hunk 1 above)
Comment (LucasKnudsen) [reply cypress/integration/brokerIsAbleToUpdateStatusOfInquiry.feature.js:26]: Yeah it's for targeting the first inquiry in the `inquiries` list and modifying that :)

File: src/components/StatusFilter.jsx (hunk 2 above)
Comment (LucasKnudsen) [reply src/components/StatusFilter.jsx:19]: A React hook that can store elements as references. In the Popover, we use this reference to anchor it to the filter button, so it doesn't pop out somewhere else random.

File: src/components/StatusFilter.jsx (hunk 3 above)
Comment (offcast-dev) [inline src/components/StatusFilter.jsx:11]: what useRef hook do? and buttonEl dost really tell me what it does

**Review by offcast-dev — APPROVED** [review]
Awesome guys! You may look into into this error stuff, but otherwise great job!

File: src/components/StatusFilter.jsx (hunk 3 above)
Comment (LucasKnudsen) [reply src/components/StatusFilter.jsx:11]: A React hook that can store elements as references. In the Popover, we use this reference to anchor it to the filter button, so it doesn't pop out somewhere else random.

Comment (LucasKnudsen) [reply src/components/StatusFilter.jsx:11]: A React hook that can store elements as references. In the Popover, we use this reference to anchor it to the filter button, so it doesn't pop out somewhere else random.

File: src/components/StatusFilter.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  const handleStatusChange = (event) => {
+    setStatus({ ...status, [event.target.name]: event.target.checked })
```
Comment (tochman) [inline src/components/StatusFilter.jsx:14]: 👍🏼

File: src/components/StatusFilter.jsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+              <Checkbox
+                checked={status.done}
+                onChange={handleStatusChange}
+                name='done'
+              />
+            }
+            label='Done'
+          />
+        </FormGroup>
+      </Popover>
```
Comment (tochman) [inline src/components/StatusFilter.jsx:58]: I would extract the entire `Popover` with children to another component

Comment (LucasKnudsen) [reply src/components/StatusFilter.jsx:58]: Besides the Popover, there's just an IconButton of 3 lines. You don't think that would be over-refactoring at this point?

Comment (tochman) [reply src/components/StatusFilter.jsx:58]: Okay, got it

**Pull request merged**